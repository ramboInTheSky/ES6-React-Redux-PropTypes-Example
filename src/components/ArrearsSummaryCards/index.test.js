import React from 'react';
import { shallow } from 'enzyme';

import { ArrearsSummaryCardsComponent } from './';

describe('<ArrearsSummaryCardsComponent />', () => {
  let props;

  beforeEach(() => {
    props = {
      loading: false,
      noDataMsg: 'there is no data sorry',
      filterLabel: 'filterLabel',
      arrears: [
        {
          createdOn: '2018-07-16T13:47:18.778Z',
          currentBalance: 100,
          openTaskCount: 12,
          tenancy: {
            address: 'A Road, Anywhere',
            name: 'Fred',
          },
          ownerOutputModel: {
            id: 'ABC123',
            name: 'Fred Bloggs',
          },
        },
        {
          createdOn: '2018-04-10T13:47:18.778Z',
          currentBalance: 50,
          openTaskCount: 2,
          tenancy: {
            address: 'Another Road, Anywhere Else',
            name: 'Jim',
          },
          ownerOutputModel: {
            id: 'DEF456',
            name: 'Jim Bob',
          },
        },
      ],
      cardLabels: {
        address: 'Address',
        currentBalance: 'Current balance',
        heading: 'Arrears',
        notificationDate: 'Notification date',
        openTasks: 'Open tasks',
        tenantName: 'Tenant name',
      },
      resultsCount: 10,
      theme: {
        colors: {
          support: {
            one: 'red',
          },
        },
      },
      userId: 'ABC123',
      youText: 'You',
      getArrearsSummary: jest.fn(),
    };
  });

  it('renders correctly with arrears', () => {
    expect(shallow(<ArrearsSummaryCardsComponent {...props} />)).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const el = shallow(<ArrearsSummaryCardsComponent {...props} />);
    el.setProps({ loading: true });
    expect(el).toMatchSnapshot();
  });

  it('renders correctly when there is no data and it is not loading', () => {
    const el = shallow(<ArrearsSummaryCardsComponent {...props} />);
    el.setProps({ arrears: [] }).update();
    expect(el).toMatchSnapshot();
  });

  it('renders correctly with no tennant details', () => {
    delete props.arrears[0].tenancy;
    expect(shallow(<ArrearsSummaryCardsComponent {...props} />)).toMatchSnapshot();
  });

  it('renders correctly with no balance', () => {
    delete props.arrears[0].currentBalance;
    expect(shallow(<ArrearsSummaryCardsComponent {...props} />)).toMatchSnapshot();
  });

  it('fires getArrearsSummary when load more is clicked', () => {
    const el = shallow(<ArrearsSummaryCardsComponent {...props} />);
    const pagination = el.find('Pagination').props();
    pagination.onNextPage({ pageNumber: 1 });
    expect(props.getArrearsSummary).toHaveBeenCalledWith({ pageNumber: 1 });
  });
});
