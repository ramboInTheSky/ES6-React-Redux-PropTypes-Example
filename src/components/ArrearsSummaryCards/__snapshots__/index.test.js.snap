// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrearsSummaryCardsComponent /> renders correctly when loading 1`] = `
<div>
  <Pagination
    items={
      Array [
        Object {
          "createdOn": "2018-07-16T13:47:18.778Z",
          "currentBalance": 100,
          "openTaskCount": 12,
          "ownerOutputModel": Object {
            "id": "ABC123",
            "name": "Fred Bloggs",
          },
          "tenancy": Object {
            "address": "A Road, Anywhere",
            "name": "Fred",
          },
        },
        Object {
          "createdOn": "2018-04-10T13:47:18.778Z",
          "currentBalance": 50,
          "openTaskCount": 2,
          "ownerOutputModel": Object {
            "id": "DEF456",
            "name": "Jim Bob",
          },
          "tenancy": Object {
            "address": "Another Road, Anywhere Else",
            "name": "Jim",
          },
        },
      ]
    }
    key="filterLabel"
    loading={true}
    onNextPage={[Function]}
    pageSize={24}
    render={[Function]}
    totalNumberOfItems={10}
  />
  <Loader
    theme={
      Object {
        "colors": Object {
          "tertiaryLighter": "#D9D4D4",
        },
      }
    }
  />
</div>
`;

exports[`<ArrearsSummaryCardsComponent /> renders correctly when there is no data and it is not loading 1`] = `
<div>
  <NoDataMessage
    message="there is no data sorry"
  />
</div>
`;

exports[`<ArrearsSummaryCardsComponent /> renders correctly with arrears 1`] = `
<div>
  <Pagination
    items={
      Array [
        Object {
          "createdOn": "2018-07-16T13:47:18.778Z",
          "currentBalance": 100,
          "openTaskCount": 12,
          "ownerOutputModel": Object {
            "id": "ABC123",
            "name": "Fred Bloggs",
          },
          "tenancy": Object {
            "address": "A Road, Anywhere",
            "name": "Fred",
          },
        },
        Object {
          "createdOn": "2018-04-10T13:47:18.778Z",
          "currentBalance": 50,
          "openTaskCount": 2,
          "ownerOutputModel": Object {
            "id": "DEF456",
            "name": "Jim Bob",
          },
          "tenancy": Object {
            "address": "Another Road, Anywhere Else",
            "name": "Jim",
          },
        },
      ]
    }
    key="filterLabel"
    loading={false}
    onNextPage={[Function]}
    pageSize={24}
    render={[Function]}
    totalNumberOfItems={10}
  />
</div>
`;

exports[`<ArrearsSummaryCardsComponent /> renders correctly with no balance 1`] = `
<div>
  <Pagination
    items={
      Array [
        Object {
          "createdOn": "2018-07-16T13:47:18.778Z",
          "openTaskCount": 12,
          "ownerOutputModel": Object {
            "id": "ABC123",
            "name": "Fred Bloggs",
          },
          "tenancy": Object {
            "address": "A Road, Anywhere",
            "name": "Fred",
          },
        },
        Object {
          "createdOn": "2018-04-10T13:47:18.778Z",
          "currentBalance": 50,
          "openTaskCount": 2,
          "ownerOutputModel": Object {
            "id": "DEF456",
            "name": "Jim Bob",
          },
          "tenancy": Object {
            "address": "Another Road, Anywhere Else",
            "name": "Jim",
          },
        },
      ]
    }
    key="filterLabel"
    loading={false}
    onNextPage={[Function]}
    pageSize={24}
    render={[Function]}
    totalNumberOfItems={10}
  />
</div>
`;

exports[`<ArrearsSummaryCardsComponent /> renders correctly with no tennant details 1`] = `
<div>
  <Pagination
    items={
      Array [
        Object {
          "createdOn": "2018-07-16T13:47:18.778Z",
          "currentBalance": 100,
          "openTaskCount": 12,
          "ownerOutputModel": Object {
            "id": "ABC123",
            "name": "Fred Bloggs",
          },
        },
        Object {
          "createdOn": "2018-04-10T13:47:18.778Z",
          "currentBalance": 50,
          "openTaskCount": 2,
          "ownerOutputModel": Object {
            "id": "DEF456",
            "name": "Jim Bob",
          },
          "tenancy": Object {
            "address": "Another Road, Anywhere Else",
            "name": "Jim",
          },
        },
      ]
    }
    key="filterLabel"
    loading={false}
    onNextPage={[Function]}
    pageSize={24}
    render={[Function]}
    totalNumberOfItems={10}
  />
</div>
`;
